File: app/src/main/java/com/whatcalendar/events/TimeSettingsReceiver.java
package com.whatcalendar.events;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.util.Log;
import com.whatcalendar.service.BackgroundService;

/* loaded from: classes.dex */
public class TimeSettingsReceiver extends BroadcastReceiver {
    private static BackgroundService.EventsChangedCallback mEventsChangedCallback;

    @Override // android.content.BroadcastReceiver
    public void onReceive(Context context, Intent intent) {
        Log.d(TimeSettingsReceiver.class.getSimpleName(), "onReceive time settings changes. Action: " + intent.getAction());
        if (mEventsChangedCallback != null) {
            mEventsChangedCallback.onTimeChanged();
        }
    }

    public static void setEventsChangedCallback(BackgroundService.EventsChangedCallback callback) {
        mEventsChangedCallback = callback;
    }
}

File: app/src/main/java/com/apollo29/calendarwatch/repository/TimeSettingsReceiver.kt
package com.apollo29.calendarwatch.repository

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import com.apollo29.calendarwatch.ble.WhatCalendarWatchManager
import com.orhanobut.logger.Logger
import dagger.hilt.android.AndroidEntryPoint
import javax.inject.Inject

@AndroidEntryPoint
class TimeSettingsReceiver : BroadcastReceiver() {
    @Inject
    lateinit var manager: WhatCalendarWatchManager

    override fun onReceive(context: Context, intent: Intent) {
        Logger.d(
            "onReceive time settings changes. Action: " + intent.action
        )
        manager.eventsChangedCallback.onTimeChanged()
    }
}